cmd/srcd/cmd/root.go

```diff
@@ -17,53 +17,30 @@ package cmd
 import (
 	"bufio"
 	"context"
-	"fmt"
-	"os"
 	"regexp"
 	"time"
 
-	"github.com/src-d/engine/cmd/srcd/config"
 	"github.com/src-d/engine/cmd/srcd/daemon"
 
-	"github.com/sirupsen/logrus"
-	"github.com/spf13/cobra"
+	"gopkg.in/src-d/go-cli.v0"
+	"gopkg.in/src-d/go-log.v1"
 )
 
-var (
-	cfgFile string
-	verbose bool
-)
+var rootCmd = cli.NewNoDefaults("srcd", "The Code as Data solution by source{d}")
+
+// Command implements the default group flags. It is meant to be embedded into
+// other application commands to provide default behavior for logging, config
+type Command struct {
+	cli.PlainCommand
+	cli.LogOptions `group:"Log Options"`
 
-// rootCmd represents the base command when called without any subcommands
-var rootCmd = &cobra.Command{
-	Use:           "srcd",
-	Short:         "The Code as Data solution by source{d}",
-	SilenceErrors: true,
```

Comment:
How does it behave now in case of error?

Reply:
I missed this, I thought we didn't change this in master. It is actually not silenced in master:
```
$ srcd init /tmp/nope
Usage:
  srcd init [workdir] [flags]

Flags:
  -h, --help   help for init

Global Flags:
      --config string   config file (default is $HOME/.srcd/config.yml)
  -v, --verbose         if true, log all of the things

path '/tmp/nope' is not a valid working directory
```

In this PR works as expected:
```

$ srcd init /tmp/nope
path '/tmp/nope' is not a valid working directory

$ srcd init --nope
unknown flag `nope'
Usage:
  srcd [OPTIONS] init [init-OPTIONS] [workdir]

Starts the daemon or restarts it if already running

Help Options:
  -h, --help                                     Show this help message

[init command options]
          --config=                              config file (default: $HOME/.srcd/config.yml)

    Log Options:
          --log-level=[info|debug|warning|error] Logging level (default: info) [$LOG_LEVEL]
          --log-format=[text|json]               log format, defaults to text on a terminal and json otherwise [$LOG_FORMAT]
          --log-fields=                          default fields for the logger, specified in json [$LOG_FIELDS]
          --log-force-format                     ignore if it is running on a terminal or not [$LOG_FORCE_FORMAT]
```